### PR TITLE
feat: log TransferProcess to ClearingHouse

### PR DESCRIPTION
**Pull request by tmberthold**
Logs a completed TransferProcess to the IDS-ClearingHouse + adds context to IDS-messages to clearinghouse.

### CODE REVIEW

File: extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/sender/LogMessageSender.java [hunk 1]
```diff
@@ -74,4 +74,25 @@ public List<Class<? extends Message>> getAllowedResponseTypes() {
...
+        jo.put("AssetId", contractAgreement.getAssetId());
+        jo.put("ContractStartDate", contractAgreement.getContractStartDate());
+        jo.put("ContractEndDate", contractAgreement.getContractEndDate());
+        jo.put("ContractSigningDate", contractAgreement.getContractSigningDate());
+        jo.put("ConsumerAgentId", contractAgreement.getConsumerAgentId());
+        jo.put("ProviderAgentId", contractAgreement.getProviderAgentId());
+        return jo.toString();
+    }
+
+    private static String buildTransferProcessPayload(TransferProcess transferProcess) {
```
Comment (efiege) [inline extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/sender/LogMessageSender.java:90]: remove static?

File: extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/sender/LogMessageSender.java [hunk 2]
```diff
@@ -49,15 +51,13 @@ public Message buildMessageHeader(LogMessage logMessage,
...
-        jo.put("ContractSigningDate", logMessage.contractAgreement().getContractSigningDate());
-        jo.put("ConsumerAgentId", logMessage.contractAgreement().getConsumerAgentId());
-        jo.put("ProviderAgentId", logMessage.contractAgreement().getProviderAgentId());
-        return jo.toString();
+        if (logMessage.eventToLog() instanceof ContractAgreement contractAgreement) {
+            return buildContractAgreementPayload(contractAgreement);
+        } else if (logMessage.eventToLog() instanceof TransferProcess transferProcess) {
+            return buildTransferProcessPayload(transferProcess);
+        } else {
+            return null;
```
Comment (efiege) [inline extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/sender/LogMessageSender.java:59]: Throw an EDCException with a message like "ObjectType {class-name} not support in LogMessageSender"

File: extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/service/IdsClearingHouseServiceImpl.java [hunk 3]
```diff
@@ -46,20 +51,22 @@ public IdsClearingHouseServiceImpl(
...
+        this.transferProcessStore = transferProcessStore;
         this.monitor = monitor;
     }
 
     @Override
     public void logContractAgreement(ContractAgreement contractAgreement, URL clearingHouseLogUrl) {
         monitor.info("Logging contract agreement to ClearingHouse");
         try {
-            var connectorBaseUrl = new URI(String.format("http://%s/", hostname.get()));
+            var connectorBaseUrl = getConnectorBaseUrl();
```
Comment (efiege) [inline extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/service/IdsClearingHouseServiceImpl.java:69]: Idea: set in constructor and replace hostname with connectorBaseUrl in the fields

Comment (efiege) [reply extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/service/IdsClearingHouseServiceImpl.java:69]: The Exception can be caught in the constructor and wrapped in an EDCException. In this way misconfiguration can already be identified on startup

**Review by efiege — CHANGES_REQUESTED** [review]

**Comment by tmberthold** [comment]
EDC can log to CH (copied request into Postman):
image

**Review by efiege — APPROVED** [review]

**Pull request merged**